nextcord/ext/commands/bot.py

```diff
@@ -236,12 +237,11 @@ def __init__(
 
     # internal helpers
 
+    # kept in since BotBase isn't a direct subclass of client and thus doesn't know
+    # about inheriting this method
     def dispatch(self, event_name: str, *args: Any, **kwargs: Any) -> None:
         # super() will resolve to Client
         super().dispatch(event_name, *args, **kwargs)  # type: ignore
-        ev = "on_" + event_name
```

Comment:
This is not included in the `Client` implementation. This is important as otherwise `extra_events` is holding `on_x` and `event_name` is just `x`.
Found by the following code not printing ready:
```py
import os

from dotenv import load_dotenv

from nextcord import Client, Intents, Message

load_dotenv()


client = Client(intents=Intents(guilds=True, guild_messages=True, message_content=True))


@client.listen()
async def on_ready():
    print("Bot is ready")


async def on_message(message: Message):
    if message.content == "ping":
        await message.channel.send("pong")
        client.remove_listener(on_message)


client.add_listener(on_message)
client.run(os.environ["TOKEN"])
```

Reply:
This problem was actually with the line below the one you commented on, as I forgot to fetch with the modified listener name (so `"on_" + event`). Fixed in 1e376a6.